buildSrc/plugins/src/main/kotlin/org/jetbrains/kotlin/KotlinNativeTest.kt

```diff
@@ -0,0 +1,436 @@
+    var doBefore: Action<in Task>? = null
+
+    @Suppress("UnstableApiUsage")
+    override fun configure(config: Closure<*>): Task {
+        super.configure(config)
+
+        // Set Gradle properties for the better navigation
+        group = LifecycleBasePlugin.VERIFICATION_GROUP
+        description = "Kotlin/Native test infrastructure task"
+
+        if (!::arguments.isInitialized) {
+            arguments = mutableListOf()
+        }
+        if (::testLogger.isInitialized && testLogger != Logger.EMPTY) {
+            arguments.add("--ktest_logger=$testLogger")
+        }
+        if (useFilter && ::source.isInitialized) {
+            arguments.add("--ktest_filter=${source.convertToPattern()}")
+        }
+        project.setDistDependencyFor(this)
+        return this
+    }
+
+    @TaskAction
+    open fun run() = project.executeAndCheck(project.file(executable).toPath(), arguments)
+
+    // Converts to runner's pattern
+    private fun String.convertToPattern() = this.replace('/', '.').replace(".kt", "") + (".*")
+
+    internal fun ProcessOutput.print(prepend: String = "") {
+        if (project.verboseTest)
+            println(prepend + """
+                |stdout:$stdOut
+                |stderr:$stdErr
+                |exit code: $exitCode
+                """.trimMargin())
+    }
+}
+
+/**
+ * Create a test task of the given type. Supports configuration with Closure passed form build.gradle file.
+ */
+fun <T: KonanTest> Project.createTest(name: String, type: Class<T>, config: Closure<*>): T =
+        project.tasks.create(name, type).apply {
+            // Apply closure set in build.gradle to get all parameters.
+            this.configure(config)
+            if (enabled) {
+                // Configure test task.
+                val target = project.testTarget
+                // If run task depends on something, compile task should also depend on this.
+                val compileTask = project.tasks.getByName("compileKonan${name.capitalize()}${target.name.capitalize()}")
+                compileTask.setSameDependenciesAs(this)
+                if (doBefore != null) compileTask.doFirst(doBefore!!)
+                compileTask.enabled = enabled
+            }
+        }
+
+/**
+ * Task to run tests compiled with TestRunner.
+ * Runs tests with GTEST output and parses it to create statistics info
+ */
+open class KonanGTest : KonanTest() {
+    init {
+        // Use GTEST logger to parse test results later
+        testLogger = Logger.GTEST
+        outputDirectory = "${project.testOutputStdlib}/$name"
+        val target = project.testTarget
+        executable = "$outputDirectory/${target.name}/$name.${target.family.exeSuffix}"
+    }
+
+    lateinit var statistics: Statistics
+
+    @TaskAction
+    override fun run() = runProcess(
+            executor = project.executor::execute,
+            executable = executable,
+            args = arguments
+    ).run {
+        statistics = parse(stdOut)
+        print()
```

Comment:
IMHO it's better to use `let { ... }` here. This `print()`  call uses implicit `this` as a receiver but at first sight it looks like a call of the top-level `kotlin.print()` function. Within the `let` block this call looks less confusing: `it.print()`